sdk/communication/azure-communication-jobrouter/azure/communication/jobrouter/_router_administration_client.py

```diff
@@ -0,0 +1,1422 @@
+                :language: python
+                :dedent: 8
+                :caption: Using a RouterAdministrationClient to update an exception policy
+        """
+        if not exception_policy_id:
+            raise ValueError("exception_policy_id cannot be None.")
+
+        patch = ExceptionPolicy(
+            name = kwargs.pop('name', exception_policy.name),
+            exception_rules = kwargs.pop('exception_rules', exception_policy.exception_rules)
+        )
+
+        return self._client.job_router_administration.upsert_exception_policy(
+            id = exception_policy_id,
+            patch = patch,
+            **kwargs
+        )
+
+    @distributed_trace
+    def get_exception_policy(
+            self,
+            exception_policy_id: str,
+            **kwargs: Any
+    ):
+        #  type: (...) -> ExceptionPolicy
+        """Retrieves an existing distribution policy by Id.
+
+        :param str exception_policy_id: Id of the policy.
+
+        :return: ExceptionPolicy
+        :rtype: ~azure.communication.jobrouter.ExceptionPolicy
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/exception_policy_crud_ops.py
+                :start-after: [START get_exception_policy]
+                :end-before: [END get_exception_policy]
+                :language: python
+                :dedent: 8
+                :caption: Using a RouterAdministrationClient to get an exception policy
+        """
+        if not exception_policy_id:
+            raise ValueError("exception_policy_id cannot be None.")
+
+        return self._client.job_router_administration.get_exception_policy(
+            id = exception_policy_id,
+            **kwargs
+        )
+
+    @distributed_trace
+    def list_exception_policies(
+            self,
+            **kwargs: Any
+    ) -> ItemPaged[ExceptionPolicyItem]:
+        """Retrieves existing exception policies.
+
+        :keyword Optional[int] results_per_page: The maximum number of results to be returned per page.
+
+        :return: An iterator like instance of ExceptionPolicyItem
+        :rtype: ~azure.core.paging.ItemPaged[~azure.communication.jobrouter.ExceptionPolicyItem]
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/exception_policy_crud_ops.py
+                :start-after: [START list_exception_policies]
+                :end-before: [END list_exception_policies]
+                :language: python
+                :dedent: 8
+                :caption: Using a RouterAdministrationClient to list exception policies
+        """
+
+        results_per_page = kwargs.pop("results_per_page", None)
+
+        params = {}
+        if results_per_page is not None:
+            params['maxpagesize'] = _SERIALIZER.query("maxpagesize", results_per_page, 'int')
```

Comment:
What are we serializing here?

Reply:
The maxpagesiz is being serialized for list ops. The current autorest version didn't generate any additional kwargs for maxpagesize. It appears all query params (at least related to paging) are being passed on as part of `params` while creatinig the `HttpRequest` https://github.com/Azure/azure-sdk-for-python/blob/restapi_job_router/repo_setup/sdk/communication/azure-communication-jobrouter/azure/communication/jobrouter/_generated/operations/_operations.py#L3217-L3247